File: 1108. Defaning an IP Address/main.java
class Solution {
    public String defangIPaddr(String address) {
        StringBuilder strBuilder = new StringBuilder();
        for(char character:address.toCharArray()){
            if (character=='.'){
                strBuilder.append("[.]");
            }
            else{
                strBuilder.append(character);
            }
        }
        return strBuilder.toString();
    }
}

File: 1108. Defaning an IP Address/main.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        result = ""
        cache = []
        for char in address:
            if char !=".":
                cache.append(char)
            else:
                new = "".join(cache)+"[.]"
                cache.clear()
                result +=new
        if cache:
            result += "".join(cache)
        return result